### DISCUSSION TITLE
Deeply nested transaction writes with Prisma

### DISCUSSION THREAD

**Discussion by NathBabs** [prisma/prisma#8637] [Q&A]
One of the advantages of Sequelize is its ability to just pass in a already structured request.body with nested relations data using includes and it'll create the relationship for you or use Mixins.

I have been using Prisma for a month now and currently i am trying to do transactions with Nested Writes , and i am finding it hard to accomplish.

I would have posted my question here, but it is too long.

Because atm i have to pull apart and build each property from the request.Body and pass it into the create for every relationship when creating with deeply nested writes.

Doing it this way, can lead to creating wrong relationships when creating.

What i want to ask is how to accomplish Nested Writes similar to Sequelize's includes where I can just pass in an already structured object with deeply nested relations from the request.body? and also

Can someone please help with how they accomplish DEEPLY (like 3 or 4 levels) nested writes ?

**Comment by ryands17** [comment]
@NathBabs 👋 

Can you share the query you perform along with any error you're facing? We support deeply nested writes as shown [here](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#nested-writes).

**Reply by NathBabs** [reply]
Hi @ryands17 

The example in the docs didn't explain how to write more that one level of relations, which it should include there. That's why i thought it would work like Sequelize, where you can just pass in a fully structured request.body using `includes`. 

I just came across a stackoverflow answer that made me realize, i had to transform the request.body and add a `create` or `createMany` property for one of the nested writes array of objects., that's how i solved it


```ts
let { flightTrips } = body.flightDetail;

flightTrips.forEach((trip, index, array) => {
            for ([key, value] of Object.entries(trip)) {
                if (key.includes('Leg')) {
                    let temp = value;
                    trip[key] =
                        { createMany: { data: temp } }
                }
            }
        })
```

i really don't know how i feel about Prisma if i have to be doing all these transfomations or destructuring of the Request.body, which was far easier with Sequelize includes or Magic Methods/Mixins

**Reply by ryands17** [reply]
Yes Prisma has a specific API for nested transactions using `create` and/or `createMany` and you would either need to transform the data in your backend or send it in the same manner that Prisma accepts from the frontend.

**Reply by NathBabs** [reply]
@ryands17 
Wow that's tough 😓, just more code, thought it'll be different with Prisma.

I think docs showing these caveats and how to use these  `create` and `createMany` with complicated examples would help.

**Reply by ryands17** [reply]
The docs have a nested `create` as I have mentioned in the link above, as long as you have relations, a `create` will work inside another `create`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
